x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -829,6 +831,30 @@ private static void checkJoin(LogicalPlan plan, Set<Failure> failures) {
                 }
             }
 
+            if (join instanceof LookupJoin) {
+                join.right().forEachDown(EsRelation.class, relation -> {
+                    var indicesWithModes = relation.index().indexNameWithModes();
+                    if (indicesWithModes.size() != 1) {
+                        failures.add(
```

Comment:
I think this check requires aliases support to be properly tested, which comes in other PR